platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBook.java

```diff
@@ -294,20 +295,25 @@ public NodeId getNextNodeId() {
      * address book transactions since genesis).
      * </p>
      *
-     * @param nextNodeId the next node ID for the address book
+     * @param newNextNodeId the next node ID for the address book
      * @return this object
      */
     @NonNull
-    public AddressBook setNextNodeId(final long nextNodeId) {
-        NodeId candidate = new NodeId(nextNodeId);
+    public AddressBook setNextNodeId(@NonNull final NodeId newNextNodeId) {
+        throwIfImmutable();
+        Objects.requireNonNull(newNextNodeId);
+        if (newNextNodeId.compareTo(this.nextNodeId) < 0) {
+            throw new IllegalArgumentException("The provided nextNodeId %s is less than the current nextNodeId %s"
+                    .formatted(newNextNodeId, this.nextNodeId));
+        }
         for (final Address address : this) {
-            if (address.getNodeId().compareTo(candidate) >= 0) {
+            if (address.getNodeId().compareTo(newNextNodeId) >= 0) {
```

Comment:
Is it necessary to validate this? The `newNextNodeId` is >= the previous `nextNodeId`, which should already be guaranteed > all existing node IDs